Assets/HoloToolkit/Utilities/Scripts/CanvasRaycastResult.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace HoloToolkit.Unity
+{
+    public struct CanvasRaycastResult : IComparable<CanvasRaycastResult>
+    {
+        public static CanvasRaycastResult Empty = new CanvasRaycastResult(new RaycastResult());
+
+        public readonly RaycastResult RaycastResult;
+        public readonly Canvas Canvas;
+
+        public CanvasRaycastResult(RaycastResult raycastResult)
+        {
+            RaycastResult = raycastResult;
+            Canvas = raycastResult.gameObject ? raycastResult.gameObject.GetComponentInParent<Canvas>() : null;
```

Comment:
I'm not sure if I understand this line quite right.

Are we checking if `raycastResult.gameObject` is null, or does this return a bool, if we've hit a `GameObject`?

Reply:
It's a null check since the struct requires an assignment to all properties